kobo/apps/openrosa/apps/logger/utils/attachment.py

```diff
@@ -0,0 +1,45 @@
+from django.db import transaction
+from django.db.models import F
+from django.db.models.query import QuerySet
+
+from kobo.apps.openrosa.apps.logger.models import Attachment, XForm
+from kobo.apps.openrosa.apps.main.models import UserProfile
+
+
+def bulk_update_attachment_storage_counters(
+    attachments: QuerySet[Attachment], subtract: bool
+):
+    """
+    Update storage counters in bulk based on a queryset of attachments
+
+    Args:
+        attachments: QuerySet of Attachment objects to process.
+        subtract: If True, subtract file size. If False, add file size.
+    """
+    user_storage_updates = {}
+    xform_storage_updates = {}
+    sign = -1 if subtract else 1
+
+    for attachment in attachments:
+        file_size = attachment.media_file_size
+        if not file_size:
+            continue
+
+        xform_id = attachment.instance.xform_id
```

Comment:
`xform_id`and `user_id` are now available on `Attachment` model let's use them instead of using joins queries.